### PR TITLE
[listeners/snmp] Update auth and priv protocols

### PR DESCRIPTION
**Pull request by AlexandreYang**
### What does this PR do?

[listeners/snmp] Update auth and priv protocols

### Motivation

Docs were out of sync with code.
See https://github.com/DataDog/datadog-agent/blob/main/pkg/collector/corechecks/snmp/gosnmplib/gosnmp_auth.go

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -782,7 +782,7 @@ api_key:
 
     ## @param privProtocol - string - optional
     ## The privacy protocol to use when connecting to your SNMP devices.
-    ## Available options are: DES, 3DES, AES, AES192, AES256, AES192C, AES256C.
+    ## Available options are: DES, AES, AES192, AES192C, AES256, AES256C
```
Comment (AlexandreYang) [inline pkg/config/config_template.yaml:785]: Note for reviewers: 3DES is only available for Python, so removing it.

3DES seems to be a workaround for:

> 		// Many vendors, including Cisco, use the 3DES key extension algorithm to extend the privacy keys that are too short when using AES,AES192 and AES256.

gosnmp seems to handle keys that are too short here:

https://github.com/gosnmp/gosnmp/pull/223/files#diff-5339dca39e1d226e550eba44e438e31157f1b05e5d937fc3133f6029051e2501R327-R331

File: cmd/agent/dist/conf.d/snmp.d/auto_conf.yaml [hunk 2]
```diff
@@ -58,7 +58,7 @@ instances:
 
     ## @param privProtocol - string - optional
     ## Privacy type to use when connecting to your SNMP devices.
-    ## It can be one of: DES, 3DES, AES, AES192, AES256, AES192C, AES256C.
```
Comment (FlorianVeaux) [inline cmd/agent/dist/conf.d/snmp.d/auto_conf.yaml:61]: Should we say that `AES` == `AES128` ? I've seen this question come up a few times

**Review by FlorianVeaux — APPROVED** [review]

File: cmd/agent/dist/conf.d/snmp.d/auto_conf.yaml (hunk 2 above)
Comment (AlexandreYang) [reply cmd/agent/dist/conf.d/snmp.d/auto_conf.yaml:61]: @FlorianVeaux Good idea. 

Added this https://github.com/DataDog/datadog-agent/pull/10233/commits/dd37c0831a79357d67f8f056db1ce88d227d1082

**Review by FlorianVeaux — APPROVED** [review]

File: cmd/agent/dist/conf.d/snmp.d/auto_conf.yaml [hunk 3]
```diff
@@ -58,7 +58,7 @@ instances:
...
+    ## Available options are: DES, AES, AES192, AES192C, AES256, AES256C
```
Comment (AlexandreYang) [inline cmd/agent/dist/conf.d/snmp.d/auto_conf.yaml:61]:
```suggestion
    ## Available options are: DES, AES (128 bits), AES192, AES192C, AES256, AES256C
```

**Pull request merged**